### PR TITLE
Board support for LILYGO T-Deck

### LINKED ISSUE #8558: LILYGO® T-Deck Support
# [LILYGO® T-Deck](https://www.lilygo.cc/products/t-deck)
LILYGO® T-Deck

## Specifications
CPU: ESP32-S3
Flash: 16MB
PSRAM: 8MB
Display: 2.8" ST7789 (320x240)
Radio: 2.4Ghz WiFi, Bluetooth 5, and LoRa (optional)

---

## Enter Flash Mode (Download Mode)

1. Press and hold the RESET button.
2. Press and hold the Trackball button (PIN 0).
3. Release the RESET button while still holding the Trackball button.
4. Release the Trackball button after 1 second.

### PR DESCRIPTION
**Pull request by rgrizzell**
ESP32-S3 based device with attached peripherals. Built-in display is working as is the SPI and I2C interfaces.

Closes #8558

### CODE REVIEW

File: ports/espressif/boards/lilygo_tdeck/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,14 @@
+USB_VID = 0x303A
+USB_PID = 0x8161
```
Comment (tannewt) [inline ports/espressif/boards/lilygo_tdeck/mpconfigboard.mk:2]: Don't just randomly change this. Get a PID assigned from Espressif. The vendor id 0x303A is theirs and they administer it here: https://github.com/espressif/usb-pids

**Review by tannewt — CHANGES_REQUESTED** [review]
Please get a proper USB PID.

File: ports/espressif/boards/lilygo_tdeck/mpconfigboard.mk (hunk 1 above)
Comment (rgrizzell) [reply ports/espressif/boards/lilygo_tdeck/mpconfigboard.mk:2]: Much appreciated. I've been looking for this for the past few days.

**Review by tannewt — APPROVED** [review]
Thank you!

**Comment by rgrizzell** [comment]
3 of 5 pin labels for the trackball were incorrect, which has been addressed. I also took the opportunity to provide more user-friendly pin labels for the Speaker, Microphone, and LoRa modules.

This should be my final change for this PR. Thanks for your patience!

File: ports/espressif/boards/lilygo_tdeck/pins.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO44) },
+
+    // SPI control pins
+    { MP_ROM_QSTR(MP_QSTR_SPI_SCK), MP_ROM_PTR(&pin_GPIO40) },
+    { MP_ROM_QSTR(MP_QSTR_SPI_MOSI), MP_ROM_PTR(&pin_GPIO41) },
+    { MP_ROM_QSTR(MP_QSTR_SPI_MISO), MP_ROM_PTR(&pin_GPIO38) },
+
+    // I2C control pins
+    { MP_ROM_QSTR(MP_QSTR_I2C_SDA), MP_ROM_PTR(&pin_GPIO18) },
+    { MP_ROM_QSTR(MP_QSTR_I2C_SCL), MP_ROM_PTR(&pin_GPIO8) },
```
Comment (tannewt) [inline ports/espressif/boards/lilygo_tdeck/pins.c:54]: I'd remove the SPI_ and I2C_ prefixes because that's how the other boards name these pins.

**Review by tannewt — CHANGES_REQUESTED** [review]
One naming suggestion.

**Review by tannewt — APPROVED** [review]
Thanks!

**Pull request merged**